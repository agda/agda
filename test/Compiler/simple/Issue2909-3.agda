{-# OPTIONS --guardedness-preserving-type-constructors #-}

open import Agda.Builtin.Coinduction
open import Agda.Builtin.IO
open import Agda.Builtin.Unit
open import Agda.Builtin.Strict

data Rec (A : ∞ Set) : Set where
  fold : ♭ A → Rec A

D : Set
D = Rec (♯ (∞ D))

d : D
d = fold (♯ d)

postulate
  returnIO : ∀ {a} {A : Set a} → A → IO A

{-# COMPILE GHC returnIO = \_ _ -> return #-}

main : IO ⊤
main = primForce d (λ _ → returnIO tt)
