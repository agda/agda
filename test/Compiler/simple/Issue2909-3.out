COMPILE_FAILED

ret > ExitFailure 1
out > Experimental feature --guardedness-preserving-type-constructors has
out > been removed.
out >
